lib/src/model.dart

```diff
@@ -3751,19 +3801,25 @@ abstract class ModelElement extends Canonicalization
 
   String get sourceFileName => element.source.fullName;
 
-  Tuple2<int, int> _lineAndColumn;
-  bool _isLineNumberComputed = false;
+  CharacterLocation _characterLocation;
+  bool _characterLocationIsSet = false;
 
   @override
-  Tuple2<int, int> get lineAndColumn {
-    // TODO(jcollins-g): implement lineAndColumn for explicit fields
-    if (!_isLineNumberComputed) {
-      _lineAndColumn = lineNumberCache.lineAndColumn(
-          element.source.fullName, element.nameOffset);
+  CharacterLocation get characterLocation {
+    if (!_characterLocationIsSet) {
+      LineInfo lineInfo = compilationUnitElement.lineInfo;
+      _characterLocationIsSet = true;
+      assert(element.nameOffset >= 0, 'Invalid location data for element: ${fullyQualifiedName}');
```

Comment:
I think you can safely remove the braces here and below?  That is just use $fullyQualifiedName...

Reply:
yes.  one day I will stop writing bash in dart.